CMakeLists.txt

```diff
@@ -162,7 +162,7 @@ if(CMAKE_COMPILER_IS_IAR)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --warn_about_c_style_casts --warnings_are_errors -Ohz")
 endif(CMAKE_COMPILER_IS_IAR)
 
-if(CMAKE_COMPILER_IS_MSVC)
+if(CMAKE_COMPILER_IS_MSVC AND NOT UNSAFE_BUILD)
```

Comment:
It's a bit confusing that the "UNSAFE_BUILD" behaviour for MSVC is configured here in mbedtls/CMakeLists.txt but in mbedtls/Library/CMakeLists.txt for the other compilers. Perhaps it would be better if all the compiler specific behaviour were moved to this file? However, I acknowledge that may have an unexpected knock-on impact on other directories.

It's also a bit strange that for other compilers --noerror is used to override --error. Would be better to only set one. That's probably more than what you want to achieve here though.

Reply:
I’ve made the experience that the best approach is to have an option ala `option(MBEDTLS_FATAL_WARNINGS “compiler warnings treated as errors’ ON)`

Then users can disable this if they have newer compilers oder whatever. In upstream ci this option should the on

You are right, this should be central managed, not as now here and there, i just did this here to not have to move much stuff around